src/middlewared/middlewared/plugins/device_/device_info_linux.py

```diff
@@ -121,6 +134,27 @@ def get_disk_details(self, block_device, disk, lshw_disks):
             if disk['size'] and disk['sectorsize']:
                 disk['blocks'] = int(disk['size'] / disk['sectorsize'])
 
+        if not disk['size'] and os.path.exists(os.path.join(disk_sys_path, 'size')):
+            with open(os.path.join(disk_sys_path, 'size'), 'r') as f:
+                disk['blocks'] = int(f.read().strip())
+            disk['size'] = disk['mediasize'] = disk['blocks'] * disk['sectorsize']
+
+        if not disk['serial']:
+            serial_cp = subprocess.Popen(
+                ['sg_vpd', '--quiet', '--page=0x80', block_device.path],
```

Comment:
Have you verified if this work with SATA? First glance makes me think its only for SCSI/SAS

Reply:
```
[server] /home/themylogin> sg_vpd --quiet --page=0x80 /dev/sde
  Unit serial number:      WD-WCAVY0211173
```

Reply:
@william-gr the above example shows it's working nicely with SATA. Thank you @themylogin  :) ( also we should already have SATA drive serial from `lshw`, this is specifically for nvme drives which don't show up in `lshw` disk class )